library-sync.sh

```diff
@@ -53,23 +67,43 @@ git add operator
 rm t.tar
 rm -rf library-master
 
+function reset_directory() {
+  git checkout HEAD -- "$1"
+  # remove any changes from the working tree in this directory that checkout left behind
+  git stash -a -- "$1"
+  git stash drop
+}
+
 SUPPORTED="ruby python nodejs perl php httpd nginx eap java webserver dotnet golang rails"
 function reset_unsupported() {
+  # there are no unsupported samples in OKD
```

Comment:
how do we define what is supported vs unsupported? We would want this to be the same for OKD and OCP. If the images are no longer maintained and supported, that should be the case for both. I don't think OKD should behave differently from OCP.

Reply:
Also if it is unsupported, why are they not removed? Wouldn't the images eventually be removed if they are not supported? wouldn't this be a problem for automation tests?

Reply:
I was basing this assumption on what was done prior to the 4.17 release. Upon your request, all of the samples from the library were updated in OKD, while only the supported ones were updated for OCP. Supported here means "receiving customer support by Red Hat".

The decision to let the unsupported samples stay in without any updates in OCP was made prior to me joining the project, so I'll let @fbm3307 or @dperaza4dustbit comment on that.

Reply:
@Prashanth684 , there are only 12 s2i samples for OCP which are being supported as per the list given above.
As for removing them, thats something of a larger effort, and since the whole samples operator is being deprecated and discussion going around for its removal, we did not remove them previously. This same has been announced to the customers also in 4.13 and 4.16 releases

Reply:
> Upon your request, all of the samples from the library were updated in OKD, while only the supported ones were updated for OCP. Supported here means "receiving customer support by Red Hat".

I do understand this stance. While I like the flexibility of OKD deciding the samples it wants to import, I'm worried about the skew between OCP and OKD samples, which might be a moot point if samples is being deprecated soon.

Reply:
in other words..i would like OCP and OKD to be in sync. In understand that this requires updates in CI tests etc..but that would prevent this need.

@fbm3307 is the samples deprecation happening anytime soon?